src/npm.rs

```diff
@@ -31,3 +31,40 @@ pub fn npm_publish(path: &str) -> Result<(), Error> {
         Ok(())
     }
 }
+
+pub fn npm_adduser(
+    registry: Option<String>,
+    scope: Option<String>,
+    always_auth: bool,
+    auth_type: Option<String>,
+) -> Result<(), Error> {
```

Comment:
You could instead do:

```rust
let args = String::new();
if let Some(registry) = registry {
    args.push_str(&format!(" --registry {}", registry));
}
```
Do that for each one to generate the optional args then

```rust
let status = Command::new("npm")
     .arg("login")
     .arg(args)
     .output()?;
```
You'll want the output instead of success so we can dump to stderr like the above functions incase things go wrong.

Let me know if this doesn't make sense

Reply:
👍 for combining the args into a string.
I tried using output here but it instantly failed the npm command, that's why I resorted to using status. It felt like output wasn't waiting for npm to do it's stdin and stdout stuff.
I'll try that again though.

Reply:
Awesome glad that helped. As for stdin failing stuff I posted a comment below about it.